kover-gradle-plugin/docs/index.md

```diff
@@ -85,223 +799,379 @@ buildscript {
-        log { /* logging config for `release` build variant */ }
     }
 }
 ```
 
-### Kotlin multiplatform project
-These are projects that use Kotlin Multiplatfrom Gradle plugin (Kotlin MPP).
-Working with such projects is identical to working with JVM and Android projects simultaneously, which are described above.
+#### Specify verification rules for named variant
+
+Configuring verification rules for named report variant `custom`
+```kotlin
+kover {
+    reports {
+        variant("custom") {
+            verify {
+                // verification rules
+            }
+        }
+    }
+}
+```
 
-Both JVM and Android targets may be present in such projects, depending on the declared targets, it is necessary to use the appropriate Kover tasks.
+You can read more about creating verification rules in the [corresponding section](#verification).
 
-If only targets with the JVM type are declared, then the [Kover default tasks](#kover-default-tasks) should be used.
+### Instrumentation
+To collect code coverage for JVM applications, Kover uses instrumentation - modification of the bytecode in order to place entry counters in certain blocks of the code.
 
-If only a target with the Android type is declared then the [Kover android tasks](#kover-android-tasks) should be used.
-In this case, starting default task like `koverHtmlReport` will result in a message `Task 'koverHtmlReport' will be skipped because no tests were executed` in the build log, and the report will not be generated.
+Instrumentation can be performed by a special JVM agent that transforms the loaded classes right while the application is running.
+This approach is called on the fly instrumentation.
+This is the approach used in the Kover Gradle Plugin.
 
-A special case is when JVM and Android targets are present in the project at the same time -
-in this case, Kover default tasks (e.g. `koverHtmlReport`) will generate reports for all JVM targets, and Kover Android tasks (e.g. `koverHtmlReportRelease`) generate reports for specific build variant.
-However, if there is a need for a single report to contain measurements for both JVM targets and for any Android build variant, [reports merging](configuring#merging-reports) can be used.
+In some cases, using JVM instrumentation agent can lead to the generation of invalid bytecode for some classes, as a result, the application may not work correctly.
+In this case, it is necessary to disable the instrumentation of the problem class. As a side effect, the coverage of such a class will not be measured, and if the excluded class was declared in the project, then its coverage in the report will always be 0.
 ```kotlin
-koverReport {
-    defaults {
-        // adds the contents of the reports of `release` Android build variant to default reports
-        mergeWith("release")
+kover {
+    currentProject {
+        instrumentation {
+            excludedClasses.add("com.example.UnInstrumented*")
+        }
     }
 }
 ```
```

Comment:
Should we also mention `disableForAll/disableForTestTasks` with a comment like `Instrumentation also may affect tests performance. In case you have some performance-sensitive scenarios, you may want to disable instrumentation conditionally, e.g., for some tasks.`